tests/test_develop.py

```diff
@@ -7,21 +7,29 @@
 import sys
 
 
+data_files_source = [
+    ("share/test.txt",),
+    ("share/test.txt",),
+    ("level1/test/test.txt",),
+    ("level1/test/test.txt",),
+]
+
+# Data file specs are tuples with (target_dir, source_dir, pattern)
 data_files_specs = [
-    ("jupyter-packaging-test", "share", "test.txt"),
-    ("jupyter-packaging-test/level1", "share", "test.txt"),
-    ("jupyter-packaging-test", "level1/test", "test.txt"),
-    ("jupyter-packaging-test/level1/level2", "level1/test", "test.txt")
+    ("jupyter-packaging-test", "share", "*.*"),
```

Comment:
I meant a full recursive check, since we won't know all of the sub folders

Reply:
Ah, I see. I'm running into issues here. This is where I see different behavior from `pip install -e .` versus `pip install .`. 

* `pip install -e .` keeps the subfolders from the recursive check in the final destination where the files are copied. 
* (I think) `pip install .` drop the subfolders from the final destination. 

I'm verifying now with some extra unit tests.